src/test/storage/table_test.cpp

```diff
@@ -89,4 +89,21 @@ TEST_F(StorageTableTest, PrivateAddSegment) {
   EXPECT_THROW(lastSegment->append("String"), std::exception);
 }
 
+TEST_F(StorageTableTest, CompressChunk) {
+  t.append({1, "Value 1"});
+  t.append({2, "Value 2"});
+
+  t.compress_chunk(ChunkID{0});
+
+  auto& compressed_chunk = t.get_chunk(ChunkID{0});
+
+  EXPECT_THROW(compressed_chunk.append({3, "Invalid Append"}), std::runtime_error);
+}
+
+TEST_F(StorageTableTest, CompressOnlyFullChunks) {
+  t.append({1, "Insufficient Row"});
+
+  EXPECT_THROW(t.compress_chunk(ChunkID{0}), std::exception);
+}
+
 }  // namespace opossum
```

Comment:
The method `const Chunk& Table::get_chunk(ChunkID chunk_id) const` is not covered in your tests.